scalafix-rules/src/main/scala/scala/meta/internal/pc/ScalafixGlobal.scala

```diff
@@ -396,8 +410,19 @@ class ScalafixGlobal(
             Iterator(lookupSymbol(name), lookupSymbol(name.otherName))
           results.flatten.filter(_ != LookupNotFound).toList match {
             case Nil =>
-              missingImports(name) = short
-              true
+              // Missing imports must be addressable via the dot operator
+              // syntax (as type projection is not allowed in imports).
+              // https://lptk.github.io/programming/2019/09/13/type-projection.html
+              if (
+                sym.isStaticMember || // Java static
+                sym.owner.ownerChain.forall { s =>
+                  // ensure the symbol can be referenced in a static manner, without any instance
+                  s.isPackageClass || s.isPackageObjectClass || s.isModule
+                }
```

Comment:
This has the same semantics as `isStatic`, but `isStatic` yields false negatives when fed with the special case `ShortName(name0.symbol.cloneSymbol(sym))` above, as (I believe) not all flags are properly set since the symbol reattached to the name does not come from the same compiler codepath.